testsupport/cleanup/clean.go

```diff
@@ -100,87 +100,96 @@ func newCleanTask(t *testing.T, cl client.Client, obj client.Object) *cleanTask
 }
 
 func (c *cleanTask) cleanObject() {
-	if c.objToClean == nil {
-		return
-	}
-	objToClean, ok := c.objToClean.DeepCopyObject().(client.Object)
-	require.True(c.t, ok)
-	userSignup, isUserSignup := c.objToClean.(*toolchainv1alpha1.UserSignup)
-	nsTemplateTier, isNsTemplateTier := c.objToClean.(*toolchainv1alpha1.NSTemplateTier)
-	kind := objToClean.GetObjectKind().GroupVersionKind().Kind
-	if kind == "" {
-		kind = reflect.TypeOf(c.objToClean).Elem().Name()
-	}
-	c.t.Logf("deleting %s: %s ...", kind, objToClean.GetName())
-	if err := c.client.Delete(context.TODO(), objToClean, propagationPolicyOpts); err != nil {
-		if errors.IsNotFound(err) {
-			// if the object was UserSignup, then let's check that the MUR was deleted as well
-			murDeleted, err := c.verifyMurDeleted(isUserSignup, userSignup, true)
-			require.NoError(c.t, err)
-			// if the object was UserSignup, then let's check that the Space was deleted as well
-			spaceDeleted, err := c.verifySpaceDeleted(isUserSignup, userSignup, true)
-			require.NoError(c.t, err)
-			// either if it was deleted or if it wasn't UserSignup, then return here
-			if murDeleted && spaceDeleted {
-				c.t.Logf("%s: %s was already deleted", kind, objToClean.GetName())
-				return
-			}
+	// only clean if test passed
+	if !c.t.Failed() {
+		if c.objToClean == nil {
+			return
```

Comment:
We can keep the following code right at the beginning of the function because there's nothing to clean up in that case
```
if c.objToClean == nil {
  return
}
```